Polynomials.Core/Polynomial.cs

```diff
@@ -0,0 +1,198 @@
+﻿using System;
+using System.Text;
+
+namespace Polynomials.Core
+{
+    public class Polynomial : IEquatable<Polynomial>, ICloneable
+    {
+        private readonly float[] _coefficients;
+
+        public float this[int i]
+        {
+            get => _coefficients[i];
+            set => _coefficients[i] = value;
+        }
+
+        public Polynomial(float[] coefficients)
+        {
+            if (coefficients.Length > 100)
+            {
+                throw new Exception("Max power must be less or equal 100");
+            }
+
+            _coefficients = coefficients;
+        }
+
+        public bool Equals(Polynomial? other)
+        {
+            if (this is null || other is null)
+            {
+                return false;
+            }
+
+            if (this._coefficients.Length != other._coefficients.Length)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < this._coefficients.Length; i++)
+            {
+                if (!FloatsAreEqualEnough(this[i], other[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        // This function made public to be able to test in xUnit
+        public static bool FloatsAreEqualEnough(float left, float right)
```

Comment:
No-no-no, please don't do that, don't make your private methods public so that you could just test them. Usually, you don't test your private methods directly. You test the ones which are public, and because they are using private methods inside, you also test private methods during that at the same time.

Reply:
Here you can use an `internal` modifier and `InternalsVisibleTo` attribute just to practice and see what that is, but please, in the future in the most of the situations you should test only functions that are meant to be public from the start.